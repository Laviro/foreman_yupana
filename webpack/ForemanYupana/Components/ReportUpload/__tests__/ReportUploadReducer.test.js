import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import {
  REPORT_UPLOAD_LOGS_POLLING,
  REPORT_UPLOAD_PROCESS_START,
  REPORT_UPLOAD_PROCESS_STOP,
  REPORT_UPLOAD_PROCESS_FINISH,
} from '../ReportUploadConstants';
import reducer from '../ReportUploadReducer';
import {
  logs,
  completed,
  processID,
  pollingProcessID,
  status,
} from '../ReportUpload.fixtures';

const fixtures = {
  'should return the initial state': {},
  'should handle REPORT_UPLOAD_LOGS_POLLING_START': {
    action: {
      type: REPORT_UPLOAD_LOGS_POLLING,
      payload: {
        pollingProcessID,
      },
    },
    'should handle REPORT_UPLOAD_LOGS_POLLING': {
      action: {
        type: REPORT_UPLOAD_LOGS_POLLING,
        payload: {
          logs,
          completed,
        },
      },
      'should handle REPORT_UPLOAD_PROCESS_START': {
        action: {
          type: REPORT_UPLOAD_PROCESS_START,
          payload: {
            processID,
            status,
          },
        },
      },
      'should handle REPORT_UPLOAD_PROCESS_STOP': {
        action: {
          type: REPORT_UPLOAD_PROCESS_STOP,
          payload: {
            logs,
            completed,
            status,
          },
        },
      },
      'should handle REPORT_UPLOAD_PROCESS_FINISH': {
        action: {
          type: REPORT_UPLOAD_PROCESS_FINISH,
          payload: {
            status,
          },
        },
      },
    },
  },
};

describe('ReportUpload reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
