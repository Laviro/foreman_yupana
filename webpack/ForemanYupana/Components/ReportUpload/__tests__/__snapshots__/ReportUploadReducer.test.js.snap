// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportUpload reducer should handle REPORT_UPLOAD_LOGS_POLLING_START 1`] = `
Object {
  "processID": 0,
  "status": "stopped",
}
`;

exports[`ReportUpload reducer should return the initial state 1`] = `
Object {
  "processID": 0,
  "status": "stopped",
}
`;
